### PR TITLE
[FL-3332] Autolock fixes

### LINKED ISSUE #3296: Inhibit automatic screen locking when connected via USB CDC
### Description of the feature you're suggesting.

Currently automatic screen lock will not be inhibited when Flipper is connected from e.g. qFlipper or PuTTY via USB CDC. This gets annoying when I need to manage files or use the console.

Most scenes seem to inhibit screen lock already so it's probably not needed to introduce an entirely new feature.

### Anything else?

_No response_

### LINKED ISSUE #3900: [Request] - Require PIN on cold boot
### Description of the feature you're suggesting.

I'm new to the Flipper and I've been loving it so far. I've set a PIN and it locks after I leave it idle for 30 seconds but what I find confusing is why the flipper does not ask for a PIN when cold booting from a poweroff.

From what I gather, you're not really supposed to fully turn off the device since the battery lasts for so long, but I really don't see a reason to have it running if I'm not using it for a whole day or more. After I power it on however, it doesn't ask for my PIN which I find to be a bit of an oversight if you want other people to not be able to use the device without your permission.

I believe this is a simple enough feature and while not urgent to implement, at least having the option to ask for PIN on poweron should be present.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by portasynthinca3**
# What's new
  - A desktop icon that shows up when a CLI session is active
  - Autolock inhibition when a CLI session is active (that includes RPC-over-USB sessions) - fixes FL-3332 and #3296
  - Always locks the device on boot if a PIN code is set irregardless of the RTC "Locked" flag - fixes #3900
  - Adds a crossed out lock icon to tell the user that autolocking is currently inhibited

# Verification 
  - Set up a PIN and 30s autolocking

Fix A:
  - While the device is unlocked, open a CLI session
  - While the device is still unlocked, press the back button to reset the inactivity timer, start a stopwatch at the same exact moment
  - Verify that the CLI session is still active after 30 seconds
  - Verify that there's a CLI icon on the desktop
  - Terminate the CLI session
  - Verify that the device locks itself after 30 seconds

Fix B:
  - Unlock the device, press the back button to reset the inactivity timer, start a stopwatch at the same exact moment
  - While the device is still unlocked, run `./fbt flash_usb_full` such that the process of uploading the files crosses the 30-second mark (the timing is tricky to get right, i admit)
  - Verify that the upload is not interrupted
  
Fix C:
  - Unlock the device, press the back button to reset the inactivity timer, start a stopwatch at the same exact moment
  - Connect to the device via qFlipper or Flipper Lab
  - While the device is still unlocked, update the firmware such that the process of uploading the files crosses the 30-second mark (again, kinda tricky)
  - Verify that the upload is not interrupted
  
Fix D:
  - Power-cycle the device using all available methods (soft reset, hard reset, reboot via settings, power off and on via settings)
  - Verify that the device locks itself upon startup no matter the state that it was in prior to it

# Checklist (For Reviewer)
  - [ ] PR has description of feature/bug or link to Confluence/Jira task
  - [ ] Description contains actions to verify feature/bugfix
  - [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by Devnol** [comment]
Lmao I actually forgot to report this other bug I found but it looks like you caught it too. Leaving the device idle while connected with RPC will autolock it and drop the RPC. The workaround I found was actually to enter any app (e.g. NFC) and just sit in that screen which prevented the autolock (dunno if that's also a bug or not, seems like intended to me).

File: applications/services/desktop/desktop_i.h [hunk 1]
```diff
@@ -73,6 +73,7 @@ struct Desktop {
     FuriPubSub* input_events_pubsub;
     FuriPubSubSubscription* input_events_subscription;
 
+    FuriSemaphore* auto_lock_inhibitors;
```
Comment (skotopes) [inline applications/services/desktop/desktop_i.h]: Looks like you don't need os level semaphore here

File: applications/services/desktop/views/desktop_events.h [hunk 2]
```diff
@@ -57,4 +57,6 @@ typedef enum {
     DesktopGlobalApiUnlock,
     DesktopGlobalSaveSettings,
     DesktopGlobalReloadSettings,
+    DesktopGlobalAddExternalInhibitor,
+    DesktopGlobalRemoveExternalInhibitor,
```
Comment (skotopes) [inline applications/services/desktop/views/desktop_events.h]: DesktopGlobalLockInhibitEnter
DesktopGlobalLockInhibitExit maybe? Same as with insomnia?

File: applications/services/cli/cli_vcp.c [hunk 3]
```diff
@@ -103,6 +121,15 @@ static int32_t vcp_worker(void* context) {
     FURI_LOG_D(TAG, "Start");
     vcp->running = true;
 
+    // GUI icon
+    vcp->desktop = furi_record_open(RECORD_DESKTOP);
```
Comment (skotopes) [inline applications/services/cli/cli_vcp.c]: image

I don't think that we need one more icon in status bar.

File: applications/services/cli/cli_vcp.c [hunk 4]
```diff
@@ -190,6 +221,10 @@ static int32_t vcp_worker(void* context) {
         }
 
         if(flags & VcpEvtStop) {
+            if(vcp->connected) {
```
Comment (skotopes) [inline applications/services/cli/cli_vcp.c]: switching between cli and any other tab in lab is not working as intended

File: applications/services/desktop/desktop.c [hunk 5]
```diff
@@ -130,16 +132,22 @@ static bool desktop_custom_event_callback(void* context, uint32_t event) {
...
         furi_semaphore_release(desktop->animation_semaphore);
 
     } else if(event == DesktopGlobalAfterAppFinished) {
         animation_manager_load_and_continue_animation(desktop->animation_manager);
-        desktop_auto_lock_arm(desktop);
+        desktop_auto_lock_remove_inhibitor(desktop);
         desktop->app_running = false;
 
+    } else if(event == DesktopGlobalAddExternalInhibitor) {
+        desktop_auto_lock_add_inhibitor(desktop);
```
Comment (skotopes) [inline applications/services/desktop/desktop.c]: enter/exit
increment/decrement

File: applications/services/desktop/desktop.c [hunk 6]
```diff
@@ -205,6 +213,24 @@ static void desktop_auto_lock_arm(Desktop* desktop) {
     }
 }
 
+static void desktop_auto_lock_add_inhibitor(Desktop* desktop) {
+    furi_check(furi_semaphore_release(desktop->auto_lock_inhibitors) == FuriStatusOk);
```
Comment (skotopes) [inline applications/services/desktop/desktop.c]: you don't need semaphore here

**Review by skotopes — CHANGES_REQUESTED** [review]

File: applications/services/cli/cli_vcp.c (hunk 3 above)
Comment (portasynthinca3) [reply applications/services/cli/cli_vcp.c]: I think it's useful to have an icon that tells the user why their device is not autolocking. Maybe the solution is to hide the CLI icon if an RPC session is active; it's kinda redundant.

File: applications/services/cli/cli_vcp.c (hunk 4 above)
Comment (portasynthinca3) [reply applications/services/cli/cli_vcp.c]: This bug has existed prior to this PR. As discussed outside of GitHub, we'll fix this later [FL-3913].

File: applications/services/cli/cli_vcp.c (hunk 3 above)
Comment (portasynthinca3) [reply applications/services/cli/cli_vcp.c]: Resolved with the UI/UX team.

File: applications/main/u2f/u2f_hid.c [hunk 7]
```diff
@@ -45,7 +45,7 @@ typedef enum {
     WorkerEvtDisconnect = (1 << 3),
     WorkerEvtRequest = (1 << 4),
     WorkerEvtUnlock = (1 << 5),
-} WorkerEvtFlags;
+} WorkerEvtFlag;
```
Comment (skotopes) [inline applications/main/u2f/u2f_hid.c:48]: Please use enum name as a base name for the items names

File: applications/services/desktop/desktop.c [hunk 8]
```diff
@@ -205,6 +219,25 @@ static void desktop_auto_lock_arm(Desktop* desktop) {
     }
 }
 
+static void desktop_auto_lock_inhibit_enter(Desktop* desktop) {
+    int32_t new_inhibitors = ++desktop->auto_lock_inhibitors;
+    FURI_LOG_D(TAG, "%ld autolock inhibitors (+1)", new_inhibitors);
+    desktop_auto_lock_inhibit(desktop);
```
Comment (skotopes) [inline applications/services/desktop/desktop.c]: why call multiple times?

**Review by skotopes — CHANGES_REQUESTED** [review]

**Comment by skotopes** [comment]
Please fix merge conflict

**Pull request closed without merging**